client/services/FeedbackGeneratorService.js

```diff
@@ -373,21 +447,58 @@ tie.factory('FeedbackGeneratorService', [
        * @returns {Feedback}
        */
       getFeedback: function(tasks, codeEvalResult, rawCodeLineIndexes) {
+        // Reset all counters but consecutiveSameErrorCounter.
+        _resetCounters('consecutiveSameError');
+
+        // If the user recieves the same error message increment the same error
+        // counter.
+        if (previousErrorString === codeEvalResult.getErrorString()) {
+          consecutiveSameErrorCounter++;
+        }
+
         var feedback = _getFeedbackWithoutReinforcement(
           tasks, codeEvalResult, rawCodeLineIndexes);
         if (tasks.length > 0 && !codeEvalResult.getErrorString()) {
           feedback.setReinforcement(
             ReinforcementGeneratorService.getReinforcement(
               tasks[tasks.length - 1], codeEvalResult));
         }
+
+        // If the same error counter reaches the threshold, prompt the user to
+        // look at the primer.
+        if (consecutiveSameErrorCounter === UNFAMILIARITY_THRESHOLD) {
+          feedback.appendTextParagraph(_getUnfamiliarLanguageFeedback());
+          consecutiveSameErrorCounter = 0;
```

Comment:
Well theoretically this could happen. We were thinking, though, if the user is getting the exact same error string (i.e. same error type with the same traceback) 5 times in a row, then odds are that either their code isn't changing or they don't understand what is the problem with the code to begin with. This is an extrapolation we made, though, and is definitely open for discussion. This is just what we came up with this when we were thinking of factors to indicate language unfamiliarity.